### ISSUE TITLE
Arrow syntax missing a closure in ES5

### ISSUE THREAD

**Issue by texastoland** [Microsoft/TypeScript#9408]
**TypeScript Version:**  1.8.0 / nightly (`2.0.0-dev.20160628`)

I'm currently using `async` in ES5 from the compiler service (similar to https://github.com/Microsoft/TypeScript/issues/1564#issuecomment-220025449) via Regenerator. Could #7706 be fixed since it's on the 2.0 roadmap?

**Code**
```js
async () => aPromise
```

**ES2015:**
```js
(() => __awaiter(this, void 0, void 0, function* () { return aPromise; }));
```
**Actual ES5:**

```js
(function () __awaiter(this, void 0, void 0, function* () { return aPromise; }));
//           ^ syntax error
```

**Expected ES5:**

```js
(function () { return __awaiter(this, void 0, void 0, function* () { return aPromise; }); });
```

CC @spicydonuts

**Comment by yortus** [comment]
Looks like #7334, which is fixed in `typescript@next`.

**Comment by texastoland** [comment]
> TypeScript Version: 1.8.0 / nightly (2.0.0-dev.20160628)

^ I've been using nightlies the entire time I've encountered it.

**Comment by yortus** [comment]
With the currently nightly, if your target is ES5, then the expression `async () => aPromise` should give you a compile-time error (`Async functions are only available when tyargering ES2015 or higher`). Are you getting this error?

The bug in v1.8 was not the invalid emitted code, it was the lack of error message. v1.8 didn't support `async` in ES5, but it didn't produce an error message if the async function was a function expression, as it is in your example here.

**Comment by texastoland** [comment]
Yes the diagnostic message is working. But the [2.0 roadmap](https://github.com/Microsoft/TypeScript/wiki/Roadmap#20) indicates planned support in #1564? I assume there's a bigger architectural theme but we've been using Regenerator with `async` successfully and I thought the missing closure might be a legitimate bug.

**Comment by yortus** [comment]
Yeah, I'd say the current emit is not a bug since there are no guarantees about emitted code when there are compile errors.

The v2.0 milestone is currently at 96% completion, so if you wait a little longer you'll be able to target ES5 and get valid emitted code. However it won't look much like your `Expected ES5` code above since that contains an ES6 generator function. I expect it will look something like the output you currently get from regenerator (ie a function implementing a state machine).

**Comment by texastoland** [comment]
Okay I understood that's the probable resolution. When I set up our build a few months ago I was silencing a number of other messages and you've already made all of the rest work. Thanks!
